From 4ea74438556ac8f4dd906e4e54e8b860eee71372 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:06 -0700
Subject: [PATCH 1630/1851] v4l: xilinx-demosaic: Fix warnings for NULL_RETURNS

commit ae1524979faa83a0e318d8d89d7fd8b1af9c7940 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the NULL_RETURNS type.

The actual warnings are:

For __xdmsc_get_pad_format():
Event return_null: 	Explicitly returning null.

For xdmsc_get_format():
Event returned_null: 	"__xdmsc_get_pad_format" returns "NULL".
Event dereference: 	Dereferencing "__xdmsc_get_pad_format(xdmsc,
cfg, fmt->pad, fmt->which)", which is known to be "NULL".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-demosaic.c   | 26 +++++++++++++++----
 1 file changed, 21 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-demosaic.c b/drivers/media/platform/xilinx/xilinx-demosaic.c
index a519c2c9719b..af464489c9e3 100644
--- a/drivers/media/platform/xilinx/xilinx-demosaic.c
+++ b/drivers/media/platform/xilinx/xilinx-demosaic.c
@@ -90,15 +90,22 @@ static struct v4l2_mbus_framefmt
 			struct v4l2_subdev_pad_config *cfg,
 			unsigned int pad, u32 which)
 {
+	struct v4l2_mbus_framefmt *get_fmt;
+
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_format(&xdmsc->xvip.subdev,
-								cfg, pad);
+		get_fmt = v4l2_subdev_get_try_format(&xdmsc->xvip.subdev,
+						     cfg, pad);
+		break;
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
-		return &xdmsc->formats[pad];
+		get_fmt = &xdmsc->formats[pad];
+		break;
 	default:
-		return NULL;
+		get_fmt = NULL;
+		break;
 	}
+
+	return get_fmt;
 }
 
 static int xdmsc_s_stream(struct v4l2_subdev *subdev, int enable)
@@ -134,8 +141,14 @@ static int xdmsc_get_format(struct v4l2_subdev *subdev,
 			    struct v4l2_subdev_format *fmt)
 {
 	struct xdmsc_dev *xdmsc = to_xdmsc(subdev);
+	struct v4l2_mbus_framefmt *get_fmt;
+
+	get_fmt = __xdmsc_get_pad_format(xdmsc, cfg, fmt->pad, fmt->which);
+	if (!get_fmt)
+		return -EINVAL;
+
+	fmt->format = *get_fmt;
 
-	fmt->format = *__xdmsc_get_pad_format(xdmsc, cfg, fmt->pad, fmt->which);
 	return 0;
 }
 
@@ -182,6 +195,9 @@ static int xdmsc_set_format(struct v4l2_subdev *subdev,
 	struct v4l2_mbus_framefmt *__format;
 
 	__format = __xdmsc_get_pad_format(xdmsc, cfg, fmt->pad, fmt->which);
+	if (!__format)
+		return -EINVAL;
+
 	*__format = fmt->format;
 
 	__format->width = clamp_t(unsigned int, fmt->format.width,
-- 
2.31.1

